ui/components/lighthouse/chat.tsx

```diff
@@ -0,0 +1,215 @@
+                  onClick={() => {
+                    append({
+                      role: "user",
+                      content: action.action,
+                    });
+                  }}
+                  className="hover:bg-muted flex h-auto w-full flex-col items-start justify-start rounded-xl border bg-gray-50 px-4 py-3.5 text-left font-sans text-sm dark:bg-gray-900"
+                >
+                  <span>{action.title}</span>
+                  <span className="text-muted-foreground">{action.label}</span>
+                </button>
+              ))}
+            </div>
+          </div>
+        </div>
+      ) : (
+        <div
+          className="flex-1 space-y-4 overflow-y-auto p-4"
+          ref={messagesContainerRef}
+        >
+          {messages.map((message, idx) => {
+            const lastUserIdx = messages
+              .map((m, i) => (m.role === "user" ? i : -1))
+              .filter((i) => i !== -1)
+              .pop();
+            const isLatestUserMsg =
+              message.role === "user" && lastUserIdx === idx;
+            return (
+              <div
+                key={message.id}
+                ref={isLatestUserMsg ? latestUserMsgRef : undefined}
+                className={`flex ${
+                  message.role === "user" ? "justify-end" : "justify-start"
+                }`}
+              >
+                <div
+                  className={`max-w-[80%] rounded-lg px-4 py-2 ${
+                    message.role === "user"
+                      ? "bg-primary text-primary-foreground dark:!text-black"
+                      : "bg-muted"
+                  }`}
+                >
+                  <div
+                    className={`prose dark:prose-invert ${message.role === "user" ? "dark:!text-black" : ""}`}
+                  >
+                    <MemoizedMarkdown
+                      id={message.id}
+                      content={message.content}
+                    />
+                  </div>
+                </div>
+              </div>
+            );
+          })}
+          {status === "submitted" && (
+            <div className="flex justify-start">
+              <div className="bg-muted max-w-[80%] rounded-lg px-4 py-2">
+                <div className="animate-pulse">Thinking...</div>
+              </div>
+            </div>
+          )}
+        </div>
+      )}
+
+      <form
+        onSubmit={handleSubmit}
+        className="mx-auto flex w-full gap-2 px-4 pb-4 md:max-w-3xl md:pb-6"
+      >
+        <div className="flex w-full items-end gap-2">
+          <textarea
+            ref={textareaRef}
+            value={input}
+            onChange={handleAutoResizeInputChange}
+            placeholder="Type your message..."
+            rows={1}
+            className="w-full flex-1 resize-none overflow-hidden rounded-lg border bg-background px-3 py-2 focus:outline-none"
+            style={{ minHeight: "40px", maxHeight: "160px" }}
```

Comment:
let's try to avoid styling `in-line` please

Reply:
This got removed after moving to CustomTextarea component.